server/service_test.go

```diff
@@ -934,3 +1192,361 @@ func TestGetPayloadToOriginRelayOnly(t *testing.T) {
+		t.Parallel()
+
+		// arrange
+		sut := newTestBackend(t, 1, time.Second)
+		defer sut.close()
+
+		sut.stubConfigurator(t, nil, sut.relaySet(t, 0))
+		sut.stubRelayGetBellatrixHeaderResponse(t, 0, 12345)
+		sut.stubRelayGetBellatrixPayloadResponse(t, 0, payload)
+		sut.requestGetHeader(t, relayHeaderPath)
+
+		// act
+		got := sut.requestGetPayload(t, relayPayloadPath, payload)
+
+		// assert
+		assertRequestWasSuccessful(t, got)
+		assertRelaysReceivedAtLeastOneRequest(t, sut)(relayPayloadPath, 0)
+	})
+
+	t.Run("Proposer has a few relays specified", func(t *testing.T) {
+		t.Parallel()
+
+		// arrange
+		sut := newTestBackend(t, 3, time.Second, withRandomRelayKeys())
+		defer sut.close()
+
+		relaysByProposer := make(map[string]relay.Set)
+		relaysByProposer[proposerPubKey.String()] = sut.relaySet(t, 0, 1, 2)
+		sut.stubConfigurator(t, relaysByProposer, nil)
+
+		sut.stubRelayGetBellatrixHeaderResponse(t, 0, 12345)
+		sut.stubRelayGetBellatrixHeaderResponse(t, 1, 45231)
+		sut.stubRelayGetBellatrixHeaderResponse(t, 2, 54321)
+
+		sut.stubRelayGetBellatrixPayloadResponse(t, 0, payload)
+		sut.stubRelayGetBellatrixPayloadResponse(t, 1, payload)
+		sut.stubRelayGetBellatrixPayloadResponse(t, 2, payload)
+		sut.requestGetHeader(t, relayHeaderPath)
+
+		// act
+		got := sut.requestGetPayload(t, relayPayloadPath, payload)
+
+		// assert
+		assertRequestWasSuccessful(t, got)
+		assertRelaysReceivedAtLeastOneRequest(t, sut)(relayPayloadPath, 0, 1, 2)
+	})
+}
+
+func BenchmarkRegisterValidator(b *testing.B) {
+	path := "/eth/v1/builder/validators"
+	reg := stubSignedValidatorRegistration()
+	payload := []types.SignedValidatorRegistration{reg}
+
+	sut := newTestBackend(b, 1, time.Second, withRandomRelayKeys(), withDiscardedOutput())
+	defer sut.close()
+
+	b.ResetTimer()
+
+	var rr *httptest.ResponseRecorder
+
+	for i := 0; i < b.N; i++ {
+		rr = sut.request(b, http.MethodPost, path, payload)
+		assertRequestWasSuccessful(b, rr)
+	}
+}
+
+func BenchmarkGetHeader(b *testing.B) {
+	parentHash := _HexToHash("0xe8b9bd82aa0e957736c5a029903e53d581edf451e28ab274f4ba314c442e35a4")
+	proposerPubKey := reltest.RandomBLSPublicKey(b)
+	relayHeaderPath := getHeaderPath(1, parentHash, proposerPubKey)
+
+	sut := newTestBackend(b, 3, time.Second, withRandomRelayKeys(), withDiscardedOutput())
+	defer sut.close()
+
+	relaysByProposer := make(map[string]relay.Set)
+	relaysByProposer[proposerPubKey.String()] = sut.relaySet(b, 0, 1, 2)
+
+	sut.stubRelayGetBellatrixHeaderResponse(b, 0, 12345)
+	sut.stubRelayGetBellatrixHeaderResponse(b, 1, 45231)
+	sut.stubRelayGetBellatrixHeaderResponse(b, 2, 54321)
+	sut.stubConfigurator(b, relaysByProposer, nil)
+
+	b.ResetTimer()
+
+	var rr *httptest.ResponseRecorder
+
+	for i := 0; i < b.N; i++ {
+		rr = sut.requestGetHeader(b, relayHeaderPath)
+		assertRequestWasSuccessful(b, rr)
+	}
+}
+
+func BenchmarkGetPayload(b *testing.B) {
+	blockHash := _HexToHash("0xc457e7cedd8bf0c16630dea852b20fb387fdb71c5ab369529ec09011371b22e1")
```

Comment:
This is done in the style of the project. I'd rather not extract the constant now. It should be done in a separate PR, as it will lead to quite a lot of changes in the current code.